gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/transformation/functions/TransformGraphTransaction.java

```diff
@@ -86,25 +96,25 @@ public TransformGraphTransaction(GraphHeadFactory<EPGMGraphHead> graphHeadFactor
   @Override
   public GraphTransaction map(GraphTransaction transaction) throws Exception {
     if (graphHeadTransFunc != null) {
-      transaction.setGraphHead(graphHeadTransFunc.apply(
-        transaction.getGraphHead(),
+      transaction.setGraphHead((EPGMGraphHead) graphHeadTransFunc.apply(
```

Comment:
This is unchecked.  Maybe check the type of the factory in the constructor to avoid ClassCastExceptions during runtime.

Reply:
The problem is the `TransformationFunction<T>`, not the factory and it is not possible to check T with `instanceof`.